rotkehlchen/api/rest.py

```diff
@@ -2674,41 +2673,64 @@ def reset_eth_staking_data(
         DBHistoryEvents(self.rotkehlchen.data.db).reset_eth_staking_data(entry_type=entry_type)
         return api_response(OK_RESULT, status_code=HTTPStatus.OK)
 
+    def _delete_evm_tx_data(
+            self,
+            tx_hash: EVMTxHash | None,
+            chain: SUPPORTED_EVM_CHAINS_TYPE | None = None,
+    ) -> None:
+        """Purge EVM transaction data from the DB."""
+        DBEvmTx(self.rotkehlchen.data.db).delete_evm_transaction_data(chain=chain, tx_hash=tx_hash)
```

Comment:
This is a useless function as you are just calling another function with same arguments. Remove this.

BUT ... please modify delete_evm_transaction_data to also take the write_cursor as argument. I explain why bit further down